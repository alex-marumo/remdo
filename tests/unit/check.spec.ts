import { getMinimizedState, loadEditorState } from "./common";
import { it } from "vitest";

//FIXME use toEqual instead of trying to name snapshots
it("check/uncheck", async ({ editor, expect, lexicalUpdate }) => {
  const notes = loadEditorState(editor, "single");
  expect(getMinimizedState(editor)).toMatchSnapshot("base");

  lexicalUpdate(() => (notes.note0.checked = true));
  expect(getMinimizedState(editor)).toMatchSnapshot("checked");

  lexicalUpdate(() => (notes.note0.checked = true));
  expect(getMinimizedState(editor)).toMatchSnapshot("base");
});

it("toggle check", async ({ editor, expect, lexicalUpdate }) => {
  const notes = loadEditorState(editor, "single");
  expect(getMinimizedState(editor)).toMatchSnapshot("base");

  lexicalUpdate(() => notes.note0.toggleChecked());
  expect(getMinimizedState(editor)).toMatchSnapshot("checked");

  lexicalUpdate(() => notes.note0.toggleChecked());
  expect(getMinimizedState(editor)).toMatchSnapshot("unchecked");
});

it("check/uncheck recursively", async ({
  editor,
  expect,
  lexicalUpdate,
}) => {
  const notes = loadEditorState(editor, "basic");
  expect(getMinimizedState(editor)).toMatchSnapshot("base");

  lexicalUpdate(() => (notes.note0.checked = true));
  expect(getMinimizedState(editor)).toMatchSnapshot("checked");

  lexicalUpdate(() => (notes.note0.checked = true));
  expect(getMinimizedState(editor)).toMatchSnapshot("base");
});

it("toggle check recursively", async ({
  editor,
  expect,
  lexicalUpdate,
}) => {
  const notes = loadEditorState(editor, "basic");
  expect(getMinimizedState(editor)).toMatchSnapshot("base");

  lexicalUpdate(() => notes.note0.toggleChecked());
  expect(getMinimizedState(editor)).toMatchSnapshot("checked");

  lexicalUpdate(() => notes.note0.toggleChecked());
  expect(getMinimizedState(editor)).toMatchSnapshot("base");
});
