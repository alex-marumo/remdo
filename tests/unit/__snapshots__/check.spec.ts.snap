// Vitest Snapshot v1

exports[`check/uncheck > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`check/uncheck > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`check/uncheck > checked 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`check/uncheck recursively > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;

exports[`check/uncheck recursively > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - checked: true
          indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;

exports[`check/uncheck recursively > checked 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - checked: true
          indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;

exports[`toggle check > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`toggle check > checked 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`toggle check > unchecked 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: false
      type: listitem
      children:
      - text: note0
        type: text
"
`;

exports[`toggle check recursively > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;

exports[`toggle check recursively > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: false
      type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - checked: false
          indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;

exports[`toggle check recursively > checked 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - checked: true
      type: listitem
      children:
      - text: note0
        type: text
    - type: listitem
      children:
      - type: list
        children:
        - checked: true
          indent: 1
          type: listitem
          children:
          - text: note1
            type: text
"
`;
